### ISSUE TITLE
Random/Flaky Unit Tests fail with `Timeout starting apiserver` in kubernetes-test-go suite

### ISSUE THREAD

**Issue by saad-ali** [kubernetes/kubernetes#20613]
From http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7518/console:

```
16:34:36 FAIL	k8s.io/kubernetes/pkg/proxy/config	300.127s
...
16:34:40 [BFA] Scanning build for known causes...
16:34:40 [BFA] Found failure cause(s):
16:34:40 [BFA] Timeout starting apiserver
16:34:40 [BFA] Done. 0s
16:34:40 [BFA] Found failed test case: TestNewMultipleSourcesServicesMultipleHandlersAddedAndNotified
````

```
Error Message: Failed

Stacktrace:
/tmp/workdir/go/src/testing/testing.go:703 +0x16b
/tmp/workdir/go/src/time/sleep.go:129 +0x6e
/tmp/workdir/go/src/testing/testing.go:562 +0xafa
/tmp/workdir/go/src/testing/testing.go:494 +0xe5
k8s.io/kubernetes/pkg/proxy/config/_test/_testmain.go:70 +0x210
/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1
/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x8c
/workspace/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2ba
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/api.go:43 +0x304
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/undelta_store.go:73 +0x18a
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:306 +0x30a
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:251 +0xa49
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:160 +0x54
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:160 +0xe5
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/api.go:56 +0x304
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/undelta_store.go:73 +0x18a
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:306 +0x30a
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:251 +0xa49
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:160 +0x54
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/client/cache/reflector.go:160 +0xe5
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:288 +0x44
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:194 +0x32c
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:288 +0x44
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:194 +0x32c
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:288 +0x44
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:194 +0x32c
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config_test.go:70 +0xbd
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config_test.go:244 +0x951
/tmp/workdir/go/src/testing/testing.go:456 +0xdd
/tmp/workdir/go/src/testing/testing.go:561 +0xaa4
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:288 +0x44
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:194 +0x32c
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:82 +0x62
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x56
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:64 +0x59
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:65 +0x7e
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/config/config.go:77 +0x349
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:209 +0x77
/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/proxy/config/config.go:213 +0xac
```

**Comment by saad-ali** [comment]
Looks like this happens on a random set of integration tests. The common symptom is `Timeout starting apiserver`
Examples:
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7536/
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7535/
  * TestNewMultipleSourcesEndpointsMultipleHandlersAddRemoveSetAndNotified
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7520/
  * TestNewServiceAddedAndNotified
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7518/
  * TestNewMultipleSourcesServicesMultipleHandlersAddedAndNotified
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7483/
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7459/
  * TestRequestExecuteRemoteCommand
* http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-test-go/7353/

**Comment by brendandburns** [comment]
This appears to be a dupe of https://github.com/kubernetes/kubernetes/issues/19435?

**Comment by saad-ali** [comment]
> This appears to be a dupe of #19435? (which was closed, but probably should be re-opened?)

Hmm, I don't think that is the same. Those tests are executed via `test-update-storage-objects.sh` with a problem in that script. These tests are run via `./hack/test-go.sh`

**Comment by saad-ali** [comment]
This may be fixed by #20657. It hasn't reoccurred in the last 4 hours since that was merged.

**Comment by saad-ali** [comment]
In the 6 hours since #20657 was merged, I have not seen the `kubernetes-test-go` fail with this issue. Closing this issue as fixed by #20657. Reopen if you see another occurrence.
